controllers/mysqlcluster_controller_test.go

```diff
@@ -76,44 +112,537 @@ spec:
+			Expect(err).ShouldNot(HaveOccurred())
+			err = k8sClient.Delete(ctx, oldSts)
+			Expect(err).ShouldNot(HaveOccurred())
+
+			serverIDBase := mathrand.Uint32()
+			cluster.Status.ServerIDBase = &serverIDBase
+			cluster.Spec.VolumeClaimTemplates = []mocov1alpha1.PersistentVolumeClaim{
 				{
-					Name:       "mysqlx",
-					Protocol:   corev1.ProtocolTCP,
-					Port:       moco.MySQLXPort,
-					TargetPort: intstr.FromInt(moco.MySQLXPort),
+					ObjectMeta: mocov1alpha1.ObjectMeta{
+						Name: mysqlDataVolumeName,
+					},
+					Spec: corev1.PersistentVolumeClaimSpec{
+						AccessModes: []corev1.PersistentVolumeAccessMode{corev1.ReadWriteOnce},
+						Resources: corev1.ResourceRequirements{
+							Requests: corev1.ResourceList{
+								corev1.ResourceStorage: *resource.NewQuantity(1<<10, resource.BinarySI),
+							},
+						},
+					},
 				},
 			}
 
+			_, err = reconciler.createOrUpdateStatefulSet(ctx, reconciler.Log, cluster)
+			Expect(err).Should(HaveOccurred())
+		})
+
+	})
+
+	Context("CronJob", func() {
+		It("should create cron job", func() {
+			isUpdated, err := reconciler.createOrUpdateCronJob(ctx, reconciler.Log, cluster)
+			Expect(err).ShouldNot(HaveOccurred())
+			Expect(isUpdated).Should(BeTrue())
+
+			job0 := &batchv1beta1.CronJob{}
+			err = k8sClient.Get(ctx, client.ObjectKey{Name: fmt.Sprintf("%s-%d", moco.UniqueName(cluster), 0), Namespace: cluster.Namespace}, job0)
+			Expect(err).ShouldNot(HaveOccurred())
+			Expect(job0).ShouldNot(BeNil())
+
+			job1 := &batchv1beta1.CronJob{}
+			err = k8sClient.Get(ctx, client.ObjectKey{Name: fmt.Sprintf("%s-%d", moco.UniqueName(cluster), 1), Namespace: cluster.Namespace}, job1)
+			Expect(err).ShouldNot(HaveOccurred())
+			Expect(job1).ShouldNot(BeNil())
+
+			isUpdated, err = reconciler.createOrUpdateCronJob(ctx, reconciler.Log, cluster)
+			Expect(err).ShouldNot(HaveOccurred())
+			Expect(isUpdated).Should(BeFalse())
+		})
+	})
+
+	Context("Services", func() {
+		It("should create services", func() {
+			isUpdated, err := reconciler.createOrUpdateServices(ctx, reconciler.Log, cluster)
+			Expect(err).ShouldNot(HaveOccurred())
+			Expect(isUpdated).Should(BeTrue())
+
+			createdPrimaryService := &corev1.Service{}
+			createdReplicaService := &corev1.Service{}
+			err = k8sClient.Get(ctx, client.ObjectKey{Name: fmt.Sprintf("%s-primary", moco.UniqueName(cluster)), Namespace: namespace}, createdPrimaryService)
+			Expect(err).ShouldNot(HaveOccurred())
+			err = k8sClient.Get(ctx, client.ObjectKey{Name: fmt.Sprintf("%s-replica", moco.UniqueName(cluster)), Namespace: namespace}, createdReplicaService)
+			Expect(err).ShouldNot(HaveOccurred())
+
+			Expect(createdPrimaryService.Spec.Ports).Should(HaveLen(2))
+			Expect(createdPrimaryService.Spec.Ports[0].Name).Should(Equal("mysql"))
+			Expect(createdPrimaryService.Spec.Ports[0].Port).Should(BeNumerically("==", moco.MySQLPort))
+
+			Expect(createdReplicaService.Spec.Ports).Should(HaveLen(2))
```

Comment:
Is it intended to only check the length of "createdReplicaService" and not its content?